src/write.rs

```diff
@@ -234,8 +233,21 @@ impl<W> WavWriter<W>
             // TODO: add the option to specify the channel mask. For now, use
             // the default assignment.
             try!(buffer.write_le_u32(channel_mask(self.spec.channels)));
-            // The field SubFormat. We use PCM audio with integer samples.
-            try!(buffer.write_all(&super::KSDATAFORMAT_SUBTYPE_PCM));
+
+            // The field SubFormat.
+            match spec.sample_format {
+                //We use PCM audio with integer samples.
+                SampleFormat::Int => try!(buffer.write_all(&super::KSDATAFORMAT_SUBTYPE_PCM)),
+
+                SampleFormat::Float => {
+                    if spec.bits_per_sample == 32 {
+                        try!(buffer.write_all(&super::KSDATAFORMAT_SUBTYPE_IEEE_FLOAT));
+                    } else {
+                        //TODO: better error condition
+                        panic!("Invalid number of bits for floating point sample.");
```

Comment:
There is still a to do in here. We should change the message to something that also indicates how to fix the problem, something along the lines of `Invalid number of bits per sample. When writing SampleFormat::Float, bits_per_sample must be 32.`.